src/TupleTools.jl

```diff
@@ -373,4 +382,14 @@ diff(v::Tuple{}) = () # similar to diff([])
 diff(v::Tuple{Any}) = ()
 diff(v::Tuple) = (v[2] - v[1], diff(Base.tail(v))...)
 
+"""
+    indexin(a::Tuple, b::Tuple)
+
+Return a tuple containing the first indices in `b` of the elements of `a`. If an element
+of `a` is not in `b`, then the corresponding index will be `nothing`.
+"""
+function indexin(a::Tuple, b::Tuple)
+    return ntuple(i -> findfirst(==(a[i]), b), length(a))
+end
```

Comment:
I guess this one is not type stable, because `findfirst` can return nothing?

One possibility would be to explicitly error when elements are missing, but then it differs from the `Base` method and we should maybe name it differently.

Reply:
Indeed, the `Base` method always returns a `Vector{Union{Nothing,Int}}`, and is therefore type stable. I checked the inferred type of this function, and it's still `NTuple{N, Union{Nothing,Int}}`, which might be fine as supposedly small type unions are ok.

Reply:
Sure. I just hope it doesn't lead to type instabilities further down in methods that use this, but I guess not.